actors/evm/src/lib.rs

```diff
@@ -312,6 +313,80 @@ fn handle_filecoin_method_input(method: u64, codec: u64, params: &[u8]) -> Vec<u
     buf
 }
 
+/// Decode the response from "filecoin_native_method". We expect:
+///
+/// 1. The exit code (u32).
+/// 2. The codec (u64).
+/// 3. The data (bytes).
+///
+/// According to the solidity ABI.
+fn handle_filecoin_method_output(mut output: Vec<u8>) -> Result<Option<IpldBlock>, ActorError> {
+    // Short-circuit if empty.
+    if output.is_empty() {
+        return Ok(None);
+    }
+
+    // TODO: This should use the "parameter" reader. But we need to move it and likely refactor
+    // something.
+    const NUM_PARAMS: usize = 3;
+    const MIN_OUTPUT: usize = 32 * NUM_PARAMS;
+    if output.len() < MIN_OUTPUT {
+        output.resize(MIN_OUTPUT, 0);
+    }
+    let exit_code: u32 = U256::from_big_endian(&output[0..32])
+        .try_into()
+        .context_code(ExitCode::USR_SERIALIZATION, "exit code not a u32")?;
+    let exit_code = ExitCode::new(exit_code);
+    let codec: u64 = U256::from_big_endian(&output[32..64])
+        .try_into()
+        .context_code(ExitCode::USR_SERIALIZATION, "returned codec not a u64")?;
+    let len_offset: usize = U256::from_big_endian(&output[64..96])
+        .try_into()
+        .context_code(ExitCode::USR_SERIALIZATION, "invalid return value offset")?;
+    let len_offset_end = len_offset
+        .checked_add(32)
+        .context_code(ExitCode::USR_SERIALIZATION, "return value offset is too large")?;
+    if len_offset_end > output.len() {
+        output.resize(len_offset_end, 0);
+    }
```

Comment:
Im pretty sure we don't pad here

Reply:
Let's leave it this way for now, and consider revisiting before mainnet.